### ISSUE TITLE
Issue with lora.py on mlx version 0.0.7 - Adapter Loading Failure

### ISSUE THREAD

**Issue by wyanzhao** [ml-explore/mlx-examples#226]
I've encountered an issue with `lora.py` when using mlx version `0.0.7`. The issue arises during the model fine-tuning process. It seems to be related to loading adapters. The issue does not occur with mlx version `0.0.6`. It seems there might be compatibility issues with the new mlx version and the adapter loading process in `lora.py`.

**Steps to Reproduce**
Install mlx version `0.0.7`.
Run `lora.py` to fine-tune the model, such as `llama2`.
Observe the error during the execution.

**Behavior**
The process raises a ValueError indicating missing parameters in various layers. Here's the traceback for reference:

```
Traceback (most recent call last):
  File "/Users/XXX/mlx-examples/lora/lora.py", line 392, in <module>
    model.load_weights(args.adapter_file)
  File "/Users/XXX/.venv/lib/python3.11/site-packages/mlx/nn/layers/base.py", line 149, in load_weights
    raise ValueError(f"Missing parameters: {missing}.")
ValueError: Missing parameters: [list of missing parameters]
```

**Comment by awni** [comment]
Actually you need to run training first `--train`. There is a more helpful error message coming in #219

**Comment by wyanzhao** [comment]
Thank you for the help. I tried the `qlora` branch [b2dc8f3](https://github.com/ml-explore/mlx-examples/commit/b2dc8f358e1ca8e3af3611647d19b49799f8074e) with MLX version 0.0.7, but the issue still persists. Here's the command I used to fine-tune the model: `python lora.py --model ./llama2-7b-mlx --train --iters 1`. Additionally, I dumped the model using command: `python convert.py --torch-path llama-2-7b --mlx-path llama2-7b-mlx`

The program still generates an error message like this: `Traceback (most recent call last):
  File "/Users/XXX/mlx-examples/lora/lora.py", line 397, in <module>
    model.load_weights(args.adapter_file)
  File "/Users/XXX/.venv/lib/python3.11/site-packages/mlx/nn/layers/base.py", line 149, in load_weights
    raise ValueError(f"Missing parameters: {missing}.")
ValueError: Missing parameters: layers.8.feed_forward.w2.weight layers.28.attention_norm.weight layers.30.feed_forward.w3.weight layers.26.attention.wv.linear.weight layers.7.attention.wv.weight layers.14.attention.wk.weight layers.9.attention.wv.weight layers.17.attention.wk.weight layers.11.attention.wq.weight layers.30.attention_norm.weight layers.30.feed_forward.w2.weight layers.6.feed_forward.w3.weight layers.17.feed_forward.w3.weight layers.2.attention.wo.weight layers.19.attention.wq.linear.weight layers.20.attention.wq.linear.weight layers.26.feed_forward.w1.weight layers.8.feed_forward.w1.weight layers.26.feed_forward.w2.weight layers.15.attention.wk.weight layers.30.ffn_norm.weight layers.19.feed_forward.w1.weight layers.15.attention_norm.weight layers.28.feed_forward.w2.weight layers.11.attention.wk.weight layers.5.feed_forward.w1.weight layers.5.feed_forward.w3.weight layers.23.attention.wk.weight layers.10.feed_forward.w1.weight layers.23.attention.wv.linear.weight layers.13.feed_forward.w2.weight layers.3.attention.wo.weight layers.17.feed_forward.w2.weight layers.21.ffn_norm.weight layers.7.feed_forward.w2.weight layers.23.feed_forward.w2.weight layers.10.attention.wq.weight layers.29.attention.wq.linear.weight layers.5.attention.wv.weight layers.2.feed_forward.w1.weight layers.24.attention.wk.weight layers.22.feed_forward.w3.weight layers.21.feed_forward.w2.weight layers.16.attention.wq.linear.weight layers.13.ffn_norm.weight layers.0.attention.wo.weight layers.15.attention.wo.weight layers.29.attention.wo.weight layers.12.feed_forward.w2.weight layers.22.attention.wk.weight layers.20.feed_forward.w3.weight layers.8.feed_forward.w3.weight layers.7.feed_forward.w1.weight layers.24.attention.wq.linear.weight layers.19.attention_norm.weight layers.9.attention.wo.weight layers.16.feed_forward.w2.weight layers.5.feed_forward.w2.weight layers.20.feed_forward.w2.weight layers.28.attention.wq.linear.weight layers.10.attention.wk.weight layers.14.attention.wv.weight layers.22.attention.wq.linear.weight layers.22.feed_forward.w2.weight layers.0.attention.wq.weight layers.7.feed_forward.w3.weight layers.15.feed_forward.w1.weight layers.14.attention_norm.weight layers.23.feed_forward.w1.weight layers.30.attention.wv.linear.weight layers.19.feed_forward.w2.weight layers.4.attention.wq.weight layers.19.attention.wk.weight layers.12.ffn_norm.weight layers.5.attention_norm.weight layers.27.feed_forward.w2.weight layers.0.attention_norm.weight layers.2.attention.wv.weight layers.26.attention_norm.weight layers.12.feed_forward.w1.weight layers.17.attention.wo.weight layers.24.attention.wv.linear.weight layers.18.ffn_norm.weight layers.7.attention.wk.weight layers.9.attention.wq.weight layers.31.feed_forward.w2.weight layers.9.feed_forward.w1.weight layers.18.attention.wv.linear.weight layers.12.attention_norm.weight layers.16.attention.wk.weight layers.25.feed_forward.w2.weight layers.11.attention.wo.weight layers.17.attention_norm.weight layers.9.attention.wk.weight layers.24.attention.wo.weight layers.2.feed_forward.w3.weight output.weight layers.31.feed_forward.w3.weight layers.29.attention.wk.weight layers.27.feed_forward.w3.weight layers.12.attention.wq.weight layers.18.feed_forward.w3.weight layers.23.attention.wq.linear.weight layers.1.feed_forward.w1.weight layers.13.attention.wv.weight layers.30.attention.wq.linear.weight layers.18.attention_norm.weight layers.6.feed_forward.w1.weight layers.6.attention.wq.weight layers.4.attention.wk.weight layers.15.ffn_norm.weight layers.2.attention.wk.weight layers.24.feed_forward.w2.weight layers.28.feed_forward.w1.weight layers.25.attention.wv.linear.weight layers.30.feed_forward.w1.weight layers.10.attention.wv.weight layers.16.ffn_norm.weight layers.31.attention.wv.linear.weight layers.20.attention.wo.weight layers.22.feed_forward.w1.weight layers.6.attention.wk.weight layers.8.attention_norm.weight layers.15.attention.wq.weight layers.16.feed_forward.w1.weight layers.10.attention_norm.weight layers.6.attention.wo.weight layers.11.attention.wv.weight layers.28.attention.wk.weight layers.7.attention.wq.weight layers.12.feed_forward.w3.weight layers.23.attention.wo.weight layers.11.feed_forward.w1.weight layers.13.attention.wq.weight layers.14.attention.wo.weight layers.0.feed_forward.w3.weight layers.25.ffn_norm.weight layers.20.attention.wk.weight layers.25.attention.wo.weight layers.26.attention.wo.weight layers.13.feed_forward.w3.weight layers.1.attention_norm.weight tok_embeddings.weight layers.19.feed_forward.w3.weight layers.4.ffn_norm.weight layers.27.attention.wq.linear.weight layers.26.attention.wq.linear.weight norm.weight layers.10.ffn_norm.weight layers.29.feed_forward.w1.weight layers.9.attention_norm.weight layers.31.attention.wk.weight layers.26.attention.wk.weight layers.18.attention.wq.linear.weight layers.6.ffn_norm.weight layers.15.feed_forward.w2.weight layers.31.attention.wo.weight layers.11.feed_forward.w3.weight layers.17.feed_forward.w1.weight layers.8.ffn_norm.weight layers.31.feed_forward.w1.weight layers.21.attention.wo.weight layers.13.attention_norm.weight layers.1.attention.wk.weight layers.26.ffn_norm.weight layers.8.attention.wk.weight layers.30.attention.wo.weight layers.31.attention.wq.linear.weight layers.28.attention.wo.weight layers.13.attention.wo.weight layers.29.ffn_norm.weight layers.3.attention_norm.weight layers.4.feed_forward.w2.weight layers.12.attention.wk.weight layers.4.attention.wv.weight layers.18.attention.wk.weight layers.28.attention.wv.linear.weight layers.0.attention.wv.weight layers.30.attention.wk.weight layers.13.attention.wk.weight layers.24.feed_forward.w3.weight layers.25.feed_forward.w1.weight layers.2.ffn_norm.weight layers.20.feed_forward.w1.weight layers.16.feed_forward.w3.weight layers.26.feed_forward.w3.weight layers.18.feed_forward.w1.weight layers.8.attention.wv.weight layers.10.feed_forward.w2.weight layers.8.attention.wq.weight layers.11.attention_norm.weight layers.3.feed_forward.w3.weight layers.24.attention_norm.weight layers.1.feed_forward.w2.weight layers.12.attention.wv.weight layers.0.feed_forward.w2.weight layers.21.attention_norm.weight layers.25.feed_forward.w3.weight layers.3.ffn_norm.weight layers.3.feed_forward.w2.weight layers.5.attention.wk.weight layers.5.attention.wq.weight layers.21.feed_forward.w1.weight layers.23.feed_forward.w3.weight layers.28.ffn_norm.weight layers.29.attention.wv.linear.weight layers.3.feed_forward.w1.weight layers.10.feed_forward.w3.weight layers.15.attention.wv.weight layers.3.attention.wv.weight layers.11.ffn_norm.weight layers.27.attention_norm.weight layers.23.ffn_norm.weight layers.7.ffn_norm.weight layers.27.attention.wk.weight layers.27.attention.wo.weight layers.10.attention.wo.weight layers.21.attention.wq.linear.weight layers.1.feed_forward.w3.weight layers.22.attention_norm.weight layers.9.feed_forward.w3.weight layers.22.ffn_norm.weight layers.31.attention_norm.weight layers.29.attention_norm.weight layers.12.attention.wo.weight layers.24.ffn_norm.weight layers.4.feed_forward.w1.weight layers.2.feed_forward.w2.weight layers.16.attention_norm.weight layers.25.attention.wq.linear.weight layers.29.feed_forward.w2.weight layers.16.attention.wv.linear.weight layers.13.feed_forward.w1.weight layers.21.attention.wk.weight layers.14.feed_forward.w2.weight layers.3.attention.wk.weight layers.23.attention_norm.weight layers.8.attention.wo.weight layers.27.ffn_norm.weight layers.7.attention_norm.weight layers.19.attention.wo.weight layers.20.attention.wv.linear.weight layers.4.feed_forward.w3.weight layers.16.attention.wo.weight layers.22.attention.wv.linear.weight layers.18.feed_forward.w2.weight layers.1.ffn_norm.weight layers.0.ffn_norm.weight layers.5.ffn_norm.weight layers.2.attention.wq.weight layers.14.feed_forward.w1.weight layers.17.ffn_norm.weight layers.28.feed_forward.w3.weight layers.22.attention.wo.weight layers.1.attention.wv.weight layers.4.attention.wo.weight layers.19.ffn_norm.weight layers.2.attention_norm.weight layers.9.feed_forward.w2.weight layers.17.attention.wv.linear.weight layers.21.attention.wv.linear.weight layers.29.feed_forward.w3.weight layers.27.attention.wv.linear.weight layers.6.feed_forward.w2.weight layers.20.ffn_norm.weight layers.14.feed_forward.w3.weight layers.4.attention_norm.weight layers.15.feed_forward.w3.weight layers.20.attention_norm.weight layers.6.attention_norm.weight layers.24.feed_forward.w1.weight layers.1.attention.wq.weight layers.18.attention.wo.weight layers.19.attention.wv.linear.weight layers.17.attention.wq.linear.weight layers.3.attention.wq.weight layers.0.attention.wk.weight layers.27.feed_forward.w1.weight layers.5.attention.wo.weight layers.25.attention.wk.weight layers.21.feed_forward.w3.weight layers.31.ffn_norm.weight layers.9.ffn_norm.weight layers.0.feed_forward.w1.weight layers.7.attention.wo.weight layers.14.ffn_norm.weight layers.6.attention.wv.weight layers.14.attention.wq.weight layers.11.feed_forward.w2.weight layers.25.attention_norm.weight layers.1.attention.wo.weight.`
